### PR TITLE
194 add cifar 10 to the summary notebook

### PR DESCRIPTION
**Pull request by kmariuszk**
This pull request adds the CIFAR-10 dataset to the relevant documentation files. The changes made are as follows:

1. Updated the summary notebook at students_documentation/summary.qmd to include information about the CIFAR-10 dataset.
2. Added a mention of the CIFAR-10 dataset in the students_documentation/README.md file.
3. Included examples in students_documentation/summary.qmd that demonstrate the usage of the CIFAR-10 dataset.
4. Provided comprehensive documentation about the CIFAR-10 dataset in the dev/tutorials/data_catalogue/ section.

Please review my changes. I am open to any feedback or suggestions you may have to further improve the implementation.

### CODE REVIEW

File: students_documentation/summary_notebook/summary.qmd
[the remark this answers is not in the document]
Comment (kmariuszk) [reply students_documentation/summary_notebook/summary.qmd:1]: Yes, it works because `cifar_10_y` returns classes of images (which are numbers from 0 to 9).

**Review by A-Salmon — APPROVED** [review]

File: students_documentation/summary_notebook/summary.qmd [hunk 1]
```diff
@@ -0,0 +1,76 @@
+# Added Features to CounterfactualExplanations.jl
+
+In this notebook, we document the new features that we have added to the CounterfactualExplanations.jl package. These features include increasing the scope of predictive models that are compatible with the package, adding unit testing and documentation, and improving the package infrastructure. We hope that this notebook will serve as a useful reference for both ourselves and the course staff.
```
Comment (RaunoArike) [inline students_documentation/summary_notebook/summary.qmd:3]: I would rephrase the second sentence as "These features include adding new counterfactual generators into the package, increasing the scope of predictive models that are compatible with the package, adding unit testing and documentation, and improving the package infrastructure," since adding generators is a very core part of what we're doing. This is a minor detail, though, and everything else looks great, so I'll approve.

**Review by RaunoArike — APPROVED** [review]
